packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -295,31 +387,26 @@ export function attemptToDispatchEvent(
         // This shouldn't happen, something went wrong but to avoid blocking
         // the whole system, dispatch the event without a target.
         // TODO: Warn.
-        targetInst = null;
+        return undefined;
       } else if (tag === HostRoot) {
         const root: FiberRoot = nearestMounted.stateNode;
         if (root.isDehydrated) {
           // If this happens during a replay something went wrong and it might block
           // the whole system.
           return getContainerFromFiber(nearestMounted);
         }
-        targetInst = null;
```

Comment:
I think we lost this case? Should also be `return undefined`

Reply:
It's covered by the `else` at the end

Reply:
good catch